arch/risc-v/src/esp32c3/esp32c3_ttyhci.c

```diff
@@ -0,0 +1,529 @@
+/****************************************************************************
+ * Name: esp_bt_tty_rxint
+ *
+ * Description:
+ *   Call to enable or disable RXRDY interrupts
+ *
+ ****************************************************************************/
+
+static void esp_bt_tty_rxint(struct uart_dev_s *dev, bool enable)
+{
+  if (enable)
+    {
+      esp_bt_hci_dev.send_avail = true;
+      esp_bt_hci_dev.dev = dev;
+    }
+  else
+    {
+      esp_bt_hci_dev.send_avail = false;
+    }
+}
+
+/****************************************************************************
+ * Name: esp_bt_tty_rxavailable
+ *
+ * Description:
+ *   Return true if the receive holding register is not empty
+ *
+ ****************************************************************************/
+
+static bool esp_bt_tty_rxavailable(struct uart_dev_s *dev)
+{
+  if (g_recv_buffer_offset < g_recv_buffer_len)
+    {
+      return true;
+    }
+  else
+    {
+      g_recv_buffer_offset = 0;
+      g_recv_buffer_len = 0;
+      memset(g_host_recv_buffer, 0x0, 1024);
+      return false;
+    }
+}
+
+/****************************************************************************
+ * Name: esp_bt_tty_txready
+ *
+ * Description:
+ * Return true if the tranmsit hardware is ready to send another byte.  This
+ * is used to determine if send() method can be called.
+ *
+ ****************************************************************************/
+
+static bool esp_bt_tty_txready(struct uart_dev_s *dev)
+{
+  if (esp_vhci_host_check_send_available())
+    {
+      return true;
+    }
+  else
+    {
+      return false;
+    }
+}
+
+/****************************************************************************
+ * Name: esp_bt_tty_txempty
+ *
+ * Description:
+ * Return true if all characters have been sent.  If for example, the UART
+ * hardware implements FIFOs, then this would mean the transmit FIFO is
+ * empty.  This method is called when the driver needs to make sure that
+ * all characters are "drained" from the TX hardware.
+ *
+ ****************************************************************************/
+
+static bool esp_bt_tty_txempty(struct uart_dev_s *dev)
+{
+  return true ;
+}
+
+/****************************************************************************
+ * Name: esp_bt_tty_send
+ *
+ * Description:
+ *
+ ****************************************************************************/
+
+static void esp_bt_tty_send(struct uart_dev_s *dev, int ch)
+{
+  /* Then send the character */
+}
+
+/****************************************************************************
+ * Name: esp_bt_send_task
+ *
+ * Description:
+ * Send the data received from controller to host
+ *
+ ****************************************************************************/
+
+static void esp_bt_send_task(void *pvparameters)
+{
+  uint8_t recv_controller_buf[CONFIG_HCI_RXBUFSIZE];
+  int32_t recv_len = 0;
+  struct uart_dev_s  *dev  = esp_bt_hci_dev.dev;
+
+  while (1)
+    {
+      usleep(10);
```

Comment:
```suggestion
      nxsig_usleep(10);
```
`usleep` should be avoided in kernel code since it modifies the `errno` and sets cancellation points.